core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -118,10 +117,7 @@ private BuddyDto getBuddy(Buddy buddyEntity)
 	{
 		if (canIncludePrivateData(buddyEntity))
```

Comment:
This `if` is no longer useful

Reply:
It still is, see the implementation of that method:
```
	static boolean canIncludePrivateData(Buddy buddyEntity)
	{
		return (buddyEntity.getReceivingStatus() == Status.ACCEPTED) || (buddyEntity.getSendingStatus() == Status.ACCEPTED);
	}
```
The private data of a user is only included in the user entity inside a buddy entity of the buddy relationship is established.

Reply:
But in the `if` and in the `else` case the code does exactly the same (`return BuddyDto.createInstance(buddyEntity);`)

Reply:
You're totally right. I didn't look at the code that's dependent on this condition.
The condition isn't necessary anymore here. The code to create the user entity based on buddy data (``BuddyUserPrivateDataDto.createInstance(Buddy)``) has the required condition already: ``if (buddyEntity.getSendingStatus() == Status.ACCEPTED)``